server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconcilerTests.java

```diff
@@ -1040,8 +1043,98 @@ public void testDoNotRebalanceToTheNodeThatNoLongerExists() {
         assertThat(allocation.routingNodes().node("data-node-2").getByShardId(shardId), notNullValue());
     }
 
+    public void testRebalanceDoesNotCauseHotSpots() {
+
+        int numberOfNodes = randomIntBetween(5, 9);
+        int shardsPerNode = randomIntBetween(4, 15);
```

Comment:
Is there any reason not to go to 1 here? I.e.:

```suggestion
        int shardsPerNode = randomIntBetween(1, 15);
```